Feat: Broker Auth + Additional Targets

YOU54F opened the pull request:
Hey @guiyomh 

Thanks for releasing this tool, I remember when you told us about it back in December, I came back to my mind and thought it would be good to get auth based support in place, it looks like you had most of the work done, just needed exposing in user flags.

I also added a load more targets as well, because rustup/cross is awesome :)

Happy to split out the changes.

There are a couple of things that will need removing prior to merging, I set the value manually in the workflow file, as it wasn't picking up the next version, and updated a load of the action tags as they were deprecated.

guiyomh commented:
@YOU54F thank for your contribution.I'll review it today. Sorry for the tartive review.

guiyomh reviewed (changes requested):
The changelog is generated by the pipeline, so you need to remove it from your PR to avoid conflicts.

Similarly, version bumping is done automatically by the pipeline. You can't predict what the version number will be, it depends on the merge order of the PRs. So leave it to the pipeline.

Otherwise, thanks for the contribution, it's great 🤩 .

guiyomh commented on CHANGELOG.md:
```diff
@@ -2,6 +2,46 @@
+- Cargo release:
+- Ring 16.20_alpha for aarch64-pc-windows-msvc + latest cross for android:
+- Default bash shell for windows runners:
+
```
Automatically generated by the CI. Please remote it from your PR.

guiyomh commented on Cargo.toml:
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "pact-graph-network"
-version = "0.7.2"
+version = "0.8.0"
```
Bumping the  version is done by the pipeline.  Please remove it from your PR.

YOU54F committed:
chore: .gitignore .DS_Store for mac contributors

YOU54F committed:
feat(ci): Use Cross to build binaries for all available targets

YOU54F committed:
feat: Support Broker Authentication Mechanisms
∙
∙ pact-graph-network now supports Pact Broker Auth

For Basic Auth users, set

∙  - PACT_BROKER_USERNAME
∙    - PACT_BROKER_PASSWORD

For Bearer Token users, set

   - PACT_BROKER_TOKEN

For options, see `--help`

Notes:- description/metadata fields set to optional. This was due to errors
seen using the https://testdemo.pactflow.io hosted broker, it may be due to invalid
contracts.

YOU54F committed:
Merge branch 'main' into main

YOU54F commented:
No rush at all @guiyomh , thanks for taking a look 👍🏾 

Changes addressed now and merged main in.

thanks buddy

guiyomh commented:
It's merged! thank for your contrib.
